csharp/src/Ice/Connection.cs

```diff
@@ -2735,7 +2734,7 @@ internal void Completed(Exception ex)
         private bool _initialized = false;
         private bool _validated = false;
 
-        private static readonly bool _compressionSupported = BZip2.Supported();
+        private static readonly bool _compressionSupported = BZip2.Supported;
```

Comment:
I would simply use `Bzip2.IsLoaded` directly, no need for this extra static field. 
And it's not obvious to me the initialization order is reliable here. `_compressionSupported` could be false even when `Bzip2.Supported` is true.